csharp/src/Ice/MultiStreamSocket.cs

```diff
@@ -281,20 +284,26 @@ internal virtual (long, long) AbortStreams(Exception exception, Func<SocketStrea
 
         internal void AddStream(long id, SocketStream stream)
         {
-            if (_streamsAborted)
-            {
-                throw new ConnectionClosedException(isClosedByPeer: false, RetryPolicy.AfterDelay(TimeSpan.Zero));
-            }
             _streams[id] = stream;
+
```

Comment:
The first `ref` below does not look correct.

Reply:
It's needed actually. It's not part of the conditional expression, it's just the "ref" keyword for the ref parameter as in `Interlocked.Increment(ref foo)`.